Classes/Controller/Form.php

```diff
@@ -1109,6 +1109,9 @@ protected function init()
 
         $this->gp = $this->utilityFuncs->getMergedGP();
 
+        if (!$this->settings['uniqueFormID']) {
+            $this->gp['randomID'] = preg_replace('/[^0-9a-z]/', '', preg_quote($this->gp['randomID']));
```

Comment:
`preg_quote` seems superfluous.

Reply:
I know it is deprecated, obsolete and unmaintained but would a backport fix of v.2.4.1 to the Formhandler v2.0.2 (last one available for T3 6.2ELTS) be to "preg_replace / quote" and "htmlspecialchars" the same lines in the corresponding files in the older version?

formhandler_2.0.2\Classes\Controller\Tx_Formhandler_Controller_Form.php
and
formhandler_2.0.2\Classes\View\Tx_Formhandler_View_Form.php

Reply:
Formhandler 2.0.2 released to TER should have them inside, or take a look by AoE https://github.com/AOEpeople/formhandler

Reply:
Yeah that's 2.0.2 but the old version with no fixes (like htmlspecialchars($name)) inside.

Reply:
Then check the forks ( https://github.com/KaffDaddy/formhandler )

Reply:
Thanks opi99 for that one. That's at least exactly what I have done:
https://github.com/KaffDaddy/formhandler/commit/366e728b81d2a7c9c83bb29e8522a8eab736df33

So it seems that is best one can do on a 6.2ELTS release.

Also thanks tyll for discovering it over 1,5years ago.